libs/utils/WordPressUtils/src/test/java/org/wordpress/android/util/LogFileCleanerTest.kt

```diff
@@ -0,0 +1,70 @@
+package org.wordpress.android.util
+
+import android.content.Context
+import android.os.Build
+import androidx.test.core.app.ApplicationProvider
+import java.io.File
+import java.io.FileReader
+import kotlin.random.Random
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import org.wordpress.android.util.helpers.logfile.LogFileCleaner
+import org.wordpress.android.util.helpers.logfile.LogFileProvider
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [Build.VERSION_CODES.O_MR1])
+class LogFileCleanerTest {
+    lateinit var logFileProvider: LogFileProvider
+    private val maxFiles = 10
+
+    @Before
+    fun setup() {
+        val context: Context = ApplicationProvider.getApplicationContext()
+        logFileProvider = LogFileProvider.fromContext(context)
+
+        repeat(maxFiles) {
+            val file = File(logFileProvider.getLogFileDirectory(), "$it.log")
+            file.writeText("$it")
+            file.setLastModified((it * 10_000L).toLong())
+        }
+
+        assert(logFileProvider.getLogFileDirectory().listFiles().count() == maxFiles)
+    }
+
+    @After
+    fun tearDown() {
+        // Delete the test directory after each test
+        logFileProvider.getLogFileDirectory().deleteRecursively()
+    }
+
+    @Test
+    fun testThatCleanerPreservesMostRecentlyCreatedFiles() {
+        LogFileCleaner(logFileProvider, 3).clean()
+
+        val remainingFileIds = logFileProvider.getLogFileDirectory().listFiles().map {
+            FileReader(it).readText()
+        }.joinToString(",") // Strings are easier to assert against than arrays
+
+        // This assertion is based on the fact that the initial list (pre-clean) would've
+        // been "0,1,2,3,4,5,6,7,8,9". Retaining the last 3 entries gives the following:
+        assertEquals("7,8,9", remainingFileIds)
```

Comment:
I personally wouldn't have made this assumption and instead rely on the `MAX_FILES`. Here is how that'd look:

```
(MAX_FILES - 1 downTo 0).take(maxLogFileCount).reversed().joinToString(",")
```

I'd also make the following changes to the top of this function as such:

```
val maxLogFileCount = 3
assert(maxLogFileCount < MAX_FILES)
LogFileCleaner(logFileProvider, maxLogFileCount).clean()
```

---

_P.S: If we were to update our Kotlin version, we could use `takeLast` instead of the whole `downTo` & `reverse` business. Unfortunately that's outside of the scope of this PR._

Reply:
Oh, the assertion `assert(maxLogFileCount < MAX_FILES)` is not necessary 🤦‍♂ We could try to preserve more than the `MAX_FILES`. Sorry about that!

Reply:
I just variabalized the whole thing, so now we'll be basically fuzzing the function on each run instead of using a static value.

Changed in 059066dc8213fc2c62b5721125e615ce7eb37337.

Reply:
Sounds good to me 👍 Some people don't like randoms in tests, but I am actually a big fan of it, so 🤐 

If you make another change in this file, do you mind squeezing a space before the `1` in `MAX_FILES -1`? 😬